node_modules/oae-authentication/lib/strategies/shibboleth/init.js

```diff
@@ -85,7 +85,17 @@ module.exports = function(config) {
             };
 
             // Get an email address from the provided headers
-            opts.email = _getBestAttributeValue(tenant.alias, 'mapEmail', headers);
+            var email = _getBestAttributeValue(tenant.alias, 'mapEmail', headers);
+
+            // Only use the email address if its valid. Don't fail the
+            // request if it's invalid
+            if (email) {
+                var validator = new Validator();
+                validator.check(email, {'code': 400, 'msg': 'Invalid email'}).isEmail();
+                if (!validator.hasErrors()) {
+                    opts.email = email;
+                }
+            }
```

Comment:
Rather than copy / paste this logic for all of these strategies, I think we can consolidate it in `AuthenticationAPI._getOrCreateUser` when `opts.authoritative === true`